api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -334,7 +344,22 @@ private void validateStringLength(String field, String fieldName, int max, int m
     }
   }
 
-  public List<Profile> listAllProfiles() {
-    return userService.getAllUsers().stream().map(this::getProfile).collect(Collectors.toList());
+  public void adminUpdateProfile(Profile updatedProfile) {
+    final DbUser user = profileMapper.profileToDbUser(updatedProfile);
+
+    // TODO RW-5104: move these DB null fixes to mapper or other common code
+
+    user.getPageVisits().forEach(dbPageVisit -> dbPageVisit.setUser(user));
+    user.getInstitutionalAffiliations().forEach(affiliation -> affiliation.setUser(user));
+    Optional.ofNullable(user.getDemographicSurvey()).ifPresent(survey -> survey.setUser(user));
+    Optional.ofNullable(user.getAddress()).ifPresent(addr -> addr.setUser(user));
+
+    // need to retain some DbUser fields which are not present in Profile
+    final DbUser existingUser = userDao.findUserByUserId(updatedProfile.getUserId());
```

Comment:
these will probably also into the mapper for RW-5104 (which needs further refinement btw)

Reply:
This just seems odd.

I believe the mapper changes are a requirement if you feel we have to do it this way. It's way too easy to have a bug where an admin change causes us to lose state in the User table (if I'm reading this right).